esrally/driver/runner.py

```diff
@@ -1533,17 +1534,14 @@ async def __call__(self, es, params):
         datafeed_id = mandatory(params, "datafeed-id", self)
         body = mandatory(params, "body", self)
         try:
-            await es.xpack.ml.put_datafeed(datafeed_id=datafeed_id, body=body)
-        except elasticsearch.TransportError as e:
+            await es.ml.put_datafeed(datafeed_id=datafeed_id, body=body)
+        except elasticsearch.BadRequestError:
             # fallback to old path
-            if e.status_code == 400:
-                await es.transport.perform_request(
-                    "PUT",
-                    f"/_xpack/ml/datafeeds/{datafeed_id}",
-                    body=body,
-                )
-            else:
-                raise e
+            await es.perform_request(
+                method="PUT",
+                path=f"/_xpack/ml/datafeeds/{datafeed_id}",
+                body=body,
+            )
```

Comment:
Can you please mention that this is still needed for Elasticsearch 6.8? That when we drop support for it we'll remove it.

And somehow mention that this applies to all other datafeeds classes.